subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/oldresult/DefaultResolvedGraphResults.java

```diff
@@ -16,20 +16,18 @@
 
 package org.gradle.api.internal.artifacts.ivyservice.resolveengine.oldresult;
 
-import org.gradle.api.artifacts.Dependency;
-
 import java.util.Map;
 
 public class DefaultResolvedGraphResults implements ResolvedGraphResults {
-    private final Map<Long, Dependency> modulesMap;
+    private final Map<Long, Object> modulesMap;
```

Comment:
This is _real_ ugly. If we make this change so that `dependency` and `constraint` are different concepts, then constraints should never end up in here.